pandas/tests/dtypes/test_dtypes.py

```diff
@@ -1243,3 +1243,15 @@ def test_loc_setitem_empty_labels_no_dtype_conversion():
 
     assert df.a.dtype == "int64"
     tm.assert_frame_equal(df, expected)
+
+
+def test_categorical_nan_no_dtype_conversion():
+    # GH 43996
+
+    df = pd.DataFrame({"a": Categorical([np.nan], [1]), "b": [1]})
+    expected = pd.DataFrame({"a": Categorical([1], [1]), "b": [1]})
+    assert df["a"].dtype == "category"
+
+    df.loc[0, "a"] = 1
```

Comment:
From the issue, this should test assigning a numpy array with an integer not just an integer

Reply:
Oh right, I'll get that changed